mobile/src/main/res/values/strings.xml

```diff
@@ -72,13 +72,13 @@
     <string name="error_network_not_available">Network is not available</string>
     <string name="error_http_connection_failed">Received an invalid response from openHAB. Please check if openHAB is working correctly (HTTP response code %d)</string>
     <string name="error_no_uuid">Unable to get openHAB UUID, connection failed</string>
-    <string name="error_invalid_url">Please enter a valid URL in a \'protocol://host:port/\' form</string>
+    <string name="error_invalid_url">Please enter a valid URL in a \'http(s)://host(:port)/\' form</string>
     <string name="error_connection_failed">Connection to host failed</string>
     <string name="error_unable_to_resolve_hostname">Unable to resolve hostname</string>
     <string name="error_connection_sslhandshake_failed">SSL Handshake failed - maybe you need a valid client certificate</string>
```

Comment:
Does it really show this error, when client cert auth fails? Doesn't this cause a http error (401, 495, 496) instead of failing the ssl handshake?

Reply:
Funny you ask. I'm currently seeing it as I disabled demo mode but didn't configure my server address. Now that I think about it: Does auto discovery look for openHAB on https and if so, how can we expect it to own a valid certificate?

Reply:
> Does auto discovery look for openHAB on https and if so, how can we expect it to own a valid certificate

Yes, and it will be changed in #562 to http. Ignoring ssl certs by default isnt good and the oh server has probably an invalid cert.

Reply:
I think the error message mueller-ma is referring to is "error_connection_sslhandshake_failed", isn't it? There're multiple different scenarios where this may happen, however, most of them has their own messages already (see https://github.com/openhab/openhab-android/blob/c31e282b764b33a8657f38c589ecb4798c1bda98/mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java#L135-L146), so this is most likely be really related to the client certificate (or, but that should be very rare, any other misconfiguration or mismatch of the server to the client (supported TLS version, support cipher suites,...). If it's related to the client cert, it depends on how the server is configured:

1) If the webserver requests a client certificate, but proceeds with the tls handshake, even if the client sends an empty certificate message (means: no client certificate), then the webserver will normally reply with a http status code (401, 200, but with an error page, ...)
2) If the webserver requests a client certificate and fails, if no valid certificate is provided by the client, a fatal handshake_failure will be send by the server, while immediately aborting the handshake. This message will be shown in this case by the app.

Reply:
There seems to be a bug.
Log: `E/OpenHABMainActivity: Error: javax.net.ssl.SSLHandshakeException: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.`
But shows the `error_connection_sslhandshake_failed` string.

Reply:
Found the issue, fix is on its way.

Reply:
See #616